frontend/src/metabase/dashboard/components/DashboardGrid.jsx

```diff
@@ -113,12 +114,29 @@ class DashboardGrid extends Component {
       selectedTabId,
     );
 
+    if (!isEditing) {
+      this.setState({
+        initialCardSizes: this.getInitialCardSizes(cards),
+      });
+    }
+
     this.setState({
       visibleCardIds,
       layouts: this.getLayouts(cards),
     });
   }
 
+  getInitialCardSizes = cards => {
+    return cards
+      .map(card => this.getLayoutForDashCard(card))
+      .reduce((acc, curr) => {
```

Comment:
nit: I'd name this something more concrete e.g. `dashcardLayout` or something. `curr` would be hard to understand without having to look around for more context.